tests/beaker_tests/cisco_interface_portchannel/test_interface_portchannel.rb

```diff
@@ -0,0 +1,335 @@
+  title_pattern:  'port-channel100',
+  manifest_props: "
+    lacp_graceful_convergence     => 'default',
+    lacp_max_bundle               => 'default',
+    lacp_min_links                => 'default',
+    lacp_suspend_individual       => 'default',
+  ",
+  code:           [0, 2],
+  resource_props: {
+    'lacp_graceful_convergence' => 'true',
+    'lacp_max_bundle'           => '16',
+    'lacp_min_links'            => '1',
+    'lacp_suspend_individual'   => 'true',
+  },
+}
+
+tests['non_default_properties_eth'] = {
+  title_pattern:  'port-channel100',
+  manifest_props: "
+    lacp_graceful_convergence     => 'false',
+    lacp_max_bundle               => '10',
+    lacp_min_links                => '3',
+    lacp_suspend_individual       => 'false',
+  ",
+  resource_props: {
+    'lacp_graceful_convergence' => 'false',
+    'lacp_max_bundle'           => '10',
+    'lacp_min_links'            => '3',
+    'lacp_suspend_individual'   => 'false',
+  },
+}
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = PUPPET_BINPATH +
+        'resource cisco_interface_portchannel port-channel100'
+  get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_interface_portchannel(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug(
+    "build_manifest_interface_portchannel :: title_pattern:\n" +
+             tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    cisco_interface_portchannel { 'port-channel100':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_interface_portchannel(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  # Build the manifest for this test
+  build_manifest_interface_portchannel(tests, id)
+
+  # test_harness_common(tests, id)
+  test_manifest(tests, id)
+  test_resource(tests, id)
+  test_idempotence(tests, id)
+
+  tests[id][:ensure] = nil
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  # -------------------------------------------------------------------
+  device = platform
+  logger.info("#### This device is of type: #{device} #####")
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+
+  if device == 'n7k'
```

Comment:
The only thing that is really changing in these groups of tests is the `id` right?  Suggest you simplify to something like this?

```
  case device
  when /n7k/
    id = 'default_properties_asym'
  when /n5k|n6k|/
    id = 'default_properties_eth'
  when /n3k|n9k/
    id = 'default_properties_sym'
  end
  tests[id][:desc] = '1.1 Default Properties'
  test_harness_interface_portchannel(tests, id)

  tests[id][:desc] = '1.2 Default Properties'
  tests[id][:ensure] = :absent
  test_harness_interface_portchannel(tests, id)
```